src/QsCompiler/Transformations/Utilities.cs

```diff
@@ -84,6 +108,27 @@ public static TypeArgsResolution GetCombinedType(TypeArgsResolution outer, TypeA
             }).Concat(outerDict.Select(x => Tuple.Create(x.Key.Item1, x.Key.Item2, x.Value))).ToImmutableArray();
         }
 
+        public static (TypedExpression, TypedExpression) GetNoOp()
```

Comment:
There is no reason for this to be public.

Reply:
As a built in operation with no arguments, there should be a built in way of generating a call to NoOp. Because this call is not context sensitive, a utility method for getting this call should be generally available to the transformations.

Reply:
I don't understand the argument here: There are a lot of library operations with no arguments. I also can't think of any transformation that would have a need to generate an identifier for NoOp. When you write code, it is always useful to keep in mind how code may evolve in the future, and make sure it can easily be extended to cover future needs. However, please don't add code to cover hypothetical scenarios that may or may not be needed in the future - we will adapt and extend the code as these needs come up.

Reply:
This is being track in a separate task.